src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/tools/ClimbingPick.java

```diff
@@ -0,0 +1,134 @@
+        }
+        for (Material mat : MaterialCollections.getAllConcretePowderColors()) {
+            materialSpeedsDef.put(mat.name(), 1D);
+        }
+        for (Material mat : MaterialCollections.getAllTerracottaColors()) {
+            materialSpeedsDef.put(mat.name(), 1D);
+        }
+
+        materialSpeedsDef.put(Material.GRAVEL.name(), 0.4);
+        materialSpeedsDef.put(Material.SAND.name(), 0.4);
+        materialSpeedsDef.put(Material.STONE.name(), 0.6);
+        materialSpeedsDef.put(Material.DIORITE.name(), 0.6);
+        materialSpeedsDef.put(Material.GRANITE.name(), 0.6);
+        materialSpeedsDef.put(Material.ANDESITE.name(), 0.6);
+        materialSpeedsDef.put(Material.NETHERRACK.name(), 0.6);
+        if (SlimefunPlugin.getMinecraftVersion().isAtLeast(MinecraftVersion.MINECRAFT_1_16)) {
+            materialSpeedsDef.put(Material.BLACKSTONE.name(), 0.6);
+        }
+        materialSpeeds = SlimefunPlugin.getItemCfg().getOrSetDefault(getID() + ".launch-amounts", materialSpeedsDef);
+    }
+    
+    @Override
+    public ItemUseHandler getItemHandler() {
+        return e -> {
+            if (!e.getClickedBlock().isPresent()) return;
+
+            Block block = e.getClickedBlock().get();
+            ItemStack item = e.getItem();
+            Player p = e.getPlayer();
+
+            if (p.getLocation().distanceSquared(block.getLocation()) > 2.25) return;
+            if (e.getClickedFace() == BlockFace.DOWN || e.getClickedFace() == BlockFace.UP) return;
+
+            if (!users.contains(p.getUniqueId())) {
+                Material mat = block.getType();
+                double launch = materialSpeeds.getOrDefault(mat.name(), 0D);
+
+                Vector velocity = new Vector(0, 0, 0);
+                if (launch > 0.05) {
+                    int efficiencyLevel = item.getEnchantments().getOrDefault(Enchantment.DIG_SPEED, 0);
+                    if (efficiencyLevel != 0){
+                        launch += (efficiencyLevel * 0.2);
+                    }
+                    velocity.setY(launch);
+
+                    users.add(p.getUniqueId());
+                    Bukkit.getScheduler().runTaskLater(SlimefunPlugin.instance, () -> users.remove(p.getUniqueId()), 4L);
+                }
+
+                ClimbingPickLaunchEvent event = new ClimbingPickLaunchEvent(p, velocity, this, item, block);
+                Bukkit.getPluginManager().callEvent(event);
+
+                if (!event.isCancelled()) {
+                    p.setVelocity(event.getVelocity());
+                    p.getWorld().playEffect(block.getLocation(), Effect.STEP_SOUND, 1);
+
+                    if (p.getGameMode() != GameMode.CREATIVE) {
+                        damageItem(p, e.getItem());
+                    }
+                }
+            }
+        };
+    }
+
+    @Override
+    public boolean isDamageable() {
+        return true;
+    }
+
+    @Override
+    public List<ItemStack> getDisplayRecipes() {
```

Comment:
When implementing this, you should also override the label method and add a new string to the language file, since it would otherwise say "These recipes can be crafted in this machine" by default.